src/search/plugins/option_parser.h

```diff
@@ -356,26 +355,23 @@ void OptionParser::add_option(
 template<typename T>
 void OptionParser::add_enum_option(
     const std::string &key,
-    const std::vector<std::string> &names,
+    const std::vector<std::pair<std::string, std::string>> &documented_names,
```

Comment:
I'm not really a fan of `documented-names`. Maybe `enum_info`, and add a short comment that each entry contains the name and its documentation?

Reply:
Fine with me, I changed it. `OptionParser::add_enum_option` will not be necessary in the new system (we can just use `add_option<T>` for an enum `T`), so the name is temporary.